### PR TITLE
implement OAS post plugin

### PR DESCRIPTION
**Pull request by jeffy-mathew**
https://tyktech.atlassian.net/browse/TT-7744

### CODE REVIEW

File: apidef/migration.go [hunk 1]
```diff
@@ -391,4 +391,16 @@ func (a *APIDefinition) SetDisabledFlags() {
 	for i := 0; i < len(a.CustomMiddleware.PostKeyAuth); i++ {
 		a.CustomMiddleware.PostKeyAuth[i].Disabled = true
 	}
+
+	if len(a.CustomMiddleware.PostKeyAuth) > 0 {
```
Comment (titpetric) [inline apidef/migration.go:395]: Here ...

File: apidef/oas/middleware.go [hunk 2]
```diff
@@ -915,3 +919,27 @@ func (p *PostAuthenticationPlugin) ExtractTo(api *apidef.APIDefinition) {
...
+
+// Fill fills PostPlugin from supplied Tyk classic api definition.
+func (p *PostPlugin) Fill(api apidef.APIDefinition) {
+	if len(api.CustomMiddleware.Post) == 0 {
+		p.Plugins = nil
+		return
+	}
+
+	p.Plugins = make(CustomPlugins, len(api.CustomMiddleware.Post))
+	p.Plugins.Fill(api.CustomMiddleware.Post)
```
Comment (titpetric) [inline apidef/oas/middleware.go:938]: ♥️

**Review by titpetric — APPROVED** [review]

File: apidef/oas/middleware.go [hunk 3]
```diff
@@ -888,6 +892,11 @@ func (p *PrePlugin) Fill(api apidef.APIDefinition) {
 
 // ExtractTo extracts PrePlugin into Tyk classic api definition.
 func (p *PrePlugin) ExtractTo(api *apidef.APIDefinition) {
+	if len(p.Plugins) == 0 {
+		api.CustomMiddleware.Pre = nil
```
Comment (furkansenharputlu) [inline apidef/oas/middleware.go:896]: I wouldn't set this one as `nil`

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4713/merge
Commit: 640bd6a4e38468ca5cc10d57d49c0c8bba02d0e9 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4060910580)

**Pull request merged**